sdk/communication/AzureCommunicationCommon/Tests/ObjC/ObjCAutoRefreshProactiveTokenCredentialTests.m

```diff
@@ -0,0 +1,296 @@
+                                                    error:nil];
+        dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(1 * NSEC_PER_SEC)), dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0), ^{
+            [credential tokenWithCompletionHandler:^(CommunicationAccessToken * _Nullable accessToken,
+                                                     NSError * _Nullable error) {
+                XCTAssertNotNil(accessToken);
+                XCTAssertNil(error);
+                XCTAssertEqual(accessToken.token, self.sampleToken);
+                XCTAssertEqual(weakSelf.fetchTokenCallCount, 1);
+                if(i == arrSize - 1){
+                    [expectation fulfill];
+                }
+                dispatch_semaphore_signal(semaphore);
+            }];
+        });
+        dispatch_semaphore_wait(semaphore, DISPATCH_TIME_FOREVER);
+    }
+    
+    [self waitForExpectationsWithTimeout:self.timeout handler:^(NSError * _Nullable error) {
+        [self failForTimeout:error testName:expectation.expectationDescription];
+    }];
+}
+
+- (void)test_FetchTokenReturnsError {
+    XCTestExpectation *expectation = [self expectationWithDescription:@"test_FetchTokenReturnsError"];
+    __weak ObjCAutoRefreshProactiveTokenCredentialTests *weakSelf = self;
+    NSString *errorDesc = @"Error while fetching token";
+    CommunicationTokenRefreshOptions *tokenRefreshOptions = [[CommunicationTokenRefreshOptions alloc]
+                                                initWithInitialToken:self.sampleExpiredToken
+                                                refreshProactively:YES
+                                                tokenRefresher:^(void (^ block)
+                                                                 (NSString * _Nullable token,
+                                                                  NSError * _Nullable error)) {
+        weakSelf.fetchTokenCallCount += 1;
+        NSDictionary *errorDictionary = @{ NSLocalizedDescriptionKey: errorDesc};
+        NSError *error = [[NSError alloc] initWithDomain:NSOSStatusErrorDomain code:400 userInfo:errorDictionary];
+        block(nil, error);
+    }];
+    CommunicationTokenCredential *credential = [[CommunicationTokenCredential alloc]
+                                                initWithOptions: tokenRefreshOptions
+                                                error:nil];
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(1 * NSEC_PER_SEC)), dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0), ^{
+        [credential tokenWithCompletionHandler:^(CommunicationAccessToken * _Nullable accessToken,
+                                                 NSError * _Nullable error) {
+            XCTAssertNotNil(error);
+            XCTAssertEqual([error.localizedDescription containsString: errorDesc], YES);
+            XCTAssertNil(accessToken);
+            XCTAssertEqual(weakSelf.fetchTokenCallCount, 2);
```

Comment:
Why is this "2"? Please remind me...

Reply:
As an expired token was passed as the initial token in the constructor refreshing token will be triggered and finalized with an error, the second time when the token API is called in line 152